Rendering/Tools/SplineInterpretation.cpp

```diff
@@ -345,6 +370,185 @@ std::vector<double> OpenInfraPlatform::UserInterface::SplineInterpretation::nume
 	return dy;
 }
 
+std::vector<SplineInterpretationElement> OpenInfraPlatform::UserInterface::SplineInterpretation::identifyElementEndpoints(
+	std::vector<std::pair<double, double>> lengthsWithCurvatures, 
+	std::vector<double> curvatureChange) const throw(...)
+{
+	// indicate the change of curvature
+	//   curvatureZero: threshold
+	//   indicator: 0 for abs(curvatureChange) <= curvatureZero
+	//              1 for abs(curvatureChange) >  curvatureZero
+	std::vector<int> indicator;
+	double curvatureZero;
+	std::tie(indicator, curvatureZero) = indicateCurvatureChange(curvatureChange);
+
+	// 'elements' holds information about element start-/end-point, length, and corresponding id in 'lengthWithCurvature'
+	std::vector<SplineInterpretationElement> elements =
+		obtainElementsFromIndicator(indicator, lengthsWithCurvatures);
+
+	// indicator of short elements will be modified to obtain long elements
+	elements = correctShortElements(elements, curvatureChange, curvatureZero);
+
+	// consecutive elements with identical indicator will be merged
+	elements = mergeShortElements(elements);
+
+	return elements;
+}
+
+std::tuple<std::vector<int>, double> OpenInfraPlatform::UserInterface::SplineInterpretation::indicateCurvatureChange(
+	std::vector<double> curvatureChange) const throw(...)
+{
+	// threshold of curvatureChange
+	const double curvatureZero = std::max(
+		0.25 * std::abs(*std::min_element(curvatureChange.begin(), curvatureChange.end())), 
+		0.25 * std::abs(*std::max_element(curvatureChange.begin(), curvatureChange.end())));
+
+	// check values of curvatureChange against threshold
+	std::vector<int> indicator(curvatureChange.size(), 0);
+	for (size_t i = 0; i < curvatureChange.size(); i++)
+		if (std::abs(curvatureChange[i]) <= curvatureZero)
+			indicator[i] = 0;
+		else
+			indicator[i] = 1;
+
+	return { indicator, curvatureZero };
+}
+
+std::vector<SplineInterpretationElement> OpenInfraPlatform::UserInterface::SplineInterpretation::obtainElementsFromIndicator(
+	std::vector<int> indicator,
+	std::vector<std::pair<double, double>> lengthsWithCurvatures) const throw(...)

```

Comment:
if you are not changing a variable within the function, it is advisable to pass it as `const &` - saves the copying time for the processor.

```suggestion
	const std::vector<int>& indicator,
	const std::vector<std::pair<double, double>>& lengthsWithCurvatures) const throw(...)
```

Primitive types (like `int` and `double`) don't need the reference `&`.

Reply:
Done commit 154ca21